stan/math/prim/meta/as_column_vector_or_scalar.hpp

```diff
@@ -33,7 +33,7 @@ inline T as_column_vector_or_scalar(const T& a) {
  * @return Same vector.
  */
 template <typename T, require_eigen_col_vector_t<T>* = nullptr>
-inline T&& as_column_vector_or_scalar(T&& a) {
+inline auto as_column_vector_or_scalar(T&& a) {
   return std::forward<T>(a);
 }
```

Comment:
@t4c1 I changed this because of what we talked about in the past where passing in an rvalue and returning an rvalue can cause the destructor to go off. Should we just have specializations for doubles where it passes back the correct reference type?

Reply:
We don't need references for doubles. Although this will always return value (even when it should return a lvalue reference). What you wanted here is `decltype(auto)`. What I would prefer here is `T` - it will behave exactly the same.